esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+    - [source](#source)
+    - [store-value](#store-value)
+    - [target](#target)
+    - [variable](#variable)
+
+<!-- /TOC -->
+
+# Language Overview
+
+ESDL Scripts are written in XML and share some similarites with XSLT, but are easier to understand and have some powerful new features. These are some comparisons and contrasts to XSLT:
+
+1. Scripts use xpaths and support standard, extended and custom xpath functions.
+2. Variables can contain scalar or nodeset values.
+3. You never have to construct a result from scratch, you are _transforming_ an existing element into your desired result. This gives sensible default behavior with no scripting and keeps required scripts smaller.
+4. Does not use the often difficult to trace template matching behavior of xslt; scripts are more easily understood procedural functions.
+
+An ESDL Script is a collection of XML elements and XPath extension functions defined in the `urn:hpcc:esdl:script` namespace, unless otherwise noted. An XML element is either an *entry point* or an *operation*.
+
+An *entry point* is a special container of ordered *operation*. As the ESP processes a transaction, it steps into defined *entry point* in turn and executes its *operations* in order. Each entry point has a *source* XML node that is its default input context, and a *target* XML node that is its default output context. Refer to the [Entry Points](#entry-points) reference for more details.
+
+As the operations are executed in each entry point, the ESDL script engine maintains an *ESDL script context* that holds state between operations, including the root-level elements corresponding to the *source* and *target* nodes of the entry points. The [Scripting Context](#scripting-context) section has additional information.
+
+## Basic Example
+
+A very common use case for scripts is transforming the incoming request to the ESP into a format required by the backend roxie service. When ESDL is used the interfaces will match, but customization can be required for legacy services, or when business logic demands that the request be modified in some way.
+
+Say we have an incoming ESP request in this format:
+
+```xml
+<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" xmlns="urn:hpccsystems:ws:wssample:testmethod">
+    <soap:Body>
+        <TestMethodRequest>
+            <Options>
+                <StartDate>
+                    <Year>1976</Year>
+                    <Month>07</Month>
+                    <Day>12</Day>
+                </StartDate>
+            </Options>
+            <SearchBy>
+                <Name>
+                    <First>Jon</First>
+                    <Last>Jonson</Last>
+                </Name>
+            </SearchBy>
+        </TestMethodRequest>
+    </soap:Body>
+</soap:Envelope>
+```
+
+When using ESDL, the default request to the backend service would look something like this (Sample 1):
```

Comment:
"When using ESDL, the" ==> "When using ESDL in front of a roxie query there is a built in default transformation which produces a call to roxie that looks something like this (Sample 1):"
"service" ==> "roxie query"?